pd/src/consensus/worker.rs

```diff
@@ -313,6 +313,7 @@ impl Worker {
                 let identity_key = current_rate.identity_key.clone();
 
                 let funding_streams = reader.funding_streams(identity_key.clone()).await?;
+                // TODO: if this rate is for an inactive validator, do we need to do anything special here regarding the next rate value?
```

Comment:
@hdevalence can you weigh in?

Reply:
Looking at the code, it looks like inactive validators will incorrectly continue having their exchange rates increase with this change. I think we need to check the status here and decide whether to update the rate or not.

Reply:
Yeah, we want the rate updates to only be done for active validators, for exactly that reason -- but I didn't think we'd implemented validator states yet?

Reply:
@hdevalence correct the state transition PR is still in progress. The state representation is available as a struct property though.

What about validators that went inactive during the block being ended? Their state hasn't yet been committed to the DB (only happens in commit_block) -- should we use the "next validator state" (which is a property of the pending block) or the "current committed validator state" (from the database)?

Reply:
Hmm, maybe we should merge this code as-is, and then in a separate chunk, refactor it to be iterating over the validators rather than iterating over the rates?